packages/core/src/core/adapter_format.ts

```diff
@@ -0,0 +1,228 @@
+    mergedElements: mergedAfterElements,
+  } = await loadElementsAndMerge(toDir, loadElementsFromFolder, adapterContext)
+  if (afterMergeErrors.length > 0) {
+    return {
+      changes: [],
+      mergeErrors: afterMergeErrors,
+      fetchErrors: [],
+      success: false,
+      updatedConfig: {},
+    }
+  }
+  const { changes } = await calcFetchChanges(
+    afterElements,
+    mergedAfterElements,
+    elementSource.createInMemoryElementSource(mergedBeforeElements),
+    await workspace.elements(false),
+    new Map([[accountName, {}]]),
+    new Set([accountName]),
+  )
+  return {
+    changes,
+    mergeErrors: [],
+    fetchErrors: [...(beforeLoadErrors ?? []), ...(afterLoadErrors ?? [])],
+    success: true,
+    updatedConfig: {},
+  }
+}
+
+type SyncWorkspaceToFolderArgs = {
+  workspace: Workspace
+  accountName: string
+  baseDir: string
+  ignoreStateElemIdMapping?: boolean
+  ignoreStateElemIdMappingForSelectors?: ElementSelector[]
+}
+
+export type SyncWorkspaceToFolderResult = {
+  errors: ReadonlyArray<SaltoError>
+}
+export const syncWorkspaceToFolder = async ({
+  workspace,
+  accountName,
+  baseDir,
+  ignoreStateElemIdMapping,
+  ignoreStateElemIdMappingForSelectors,
+}: SyncWorkspaceToFolderArgs): Promise<SyncWorkspaceToFolderResult> => {
+  const { resolvedElements, adapter, adapterContext } = await getAdapterAndContext({
+    workspace,
+    accountName,
+    ignoreStateElemIdMapping,
+    ignoreStateElemIdMappingForSelectors,
+  })
+  const { loadElementsFromFolder, dumpElementsToFolder } = adapter
+  if (loadElementsFromFolder === undefined) {
+    throw new Error(`Account ${accountName}'s adapter does not support loading a non-nacl format`)
+  }
+  if (dumpElementsToFolder === undefined) {
+    throw new Error(`Account ${accountName}'s adapter does not support writing a non-nacl format`)
+  }
+
+  log.debug('Loading elements from folder %s', baseDir)
+  const loadResult = await loadElementsAndMerge(baseDir, loadElementsFromFolder, adapterContext)
+  if (!_.isEmpty(loadResult.loadErrors) || !_.isEmpty(loadResult.mergeErrors)) {
+    return {
+      errors: makeArray(loadResult.loadErrors).concat(loadResult.mergeErrors.map(mergeError => mergeError.error)),
+    }
+  }
+
+  const workspaceElementIDs = new Set(resolvedElements.map(elem => elem.elemID.getFullName()))
+
+  const deleteChanges = loadResult.elements
+    .filter(elem => !workspaceElementIDs.has(elem.elemID.getFullName()))
+    .map(elem => toChange({ before: elem as ChangeDataType }))
+
+  const folderElementsByID = _.keyBy(loadResult.elements, elem => elem.elemID.getFullName())
+
+  const changes = deleteChanges.concat(
+    resolvedElements.map(elem =>
+      toChange({
+        before: folderElementsByID[elem.elemID.getFullName()] as ChangeDataType,
+        after: elem as ChangeDataType,
+      }),
+    ),
+  )
+
+  log.debug(
+    'Loaded %d elements from folder %s, applying %d delete changes and %d addition changes to folder %s',
```

Comment:
```suggestion
    'Loaded %d elements from folder %s, applying %d delete changes and %d addition(or modification) changes to folder %s',
``` 

just to make sure I understand this correctly.